### PR TITLE
fold: Handle input with newlines

### PR DESCRIPTION
**Pull request by jaggededgedjustice**
The [read_line](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line) function appends to the given buffer which is never cleared. This leads to lins being duplicated, for example the following input 
```
A
B
```
Would produce the following output
```
A
AB
```

### CODE REVIEW

File: src/uu/fold/src/fold.rs [hunk 1]
```diff
@@ -98,8 +98,9 @@ fn fold(filenames: Vec<String>, bytes: bool, spaces: bool, width: usize) {
 
 #[inline]
 fn fold_file<T: Read>(mut file: BufReader<T>, bytes: bool, spaces: bool, width: usize) {
-    let mut line = String::new();
-    while safe_unwrap!(file.read_line(&mut line)) > 0 {
+    let file_lines = file.lines();
+    for line_result in file_lines {
```
Comment (sylvestre) [inline src/uu/fold/src/fold.rs:102]:
```suggestion
    for line_result in file.lines() {
```
The temporary variable doesn't seem to be used.

**Review by sylvestre — COMMENTED** [review]
Looks good.
Could you please add a second test showing the error and make sure that it doesn't happen again?
 (non regression test)

**Comment by jaggededgedjustice** [comment]
Sorry, I'm not sure what you want for the second test. The test I've added will fail with the current code and catch a regression.

File: src/uu/fold/src/fold.rs [hunk 2]
```diff
@@ -98,8 +98,8 @@ fn fold(filenames: Vec<String>, bytes: bool, spaces: bool, width: usize) {
 
 #[inline]
 fn fold_file<T: Read>(mut file: BufReader<T>, bytes: bool, spaces: bool, width: usize) {
-    let mut line = String::new();
-    while safe_unwrap!(file.read_line(&mut line)) > 0 {
+    for line_result in file.lines() {
+        let mut line = safe_unwrap!(line_result);
```
Comment (sylvestre) [inline src/uu/fold/src/fold.rs:102]: Seems that the compiler is still complaining about this line

Comment (jaggededgedjustice) [reply src/uu/fold/src/fold.rs:102]: What are you seeing? Only complaint I saw was a warning about the `file` arg not needing to be `mut`, which I've fixed and with that a compile emits no warnings.

Comment (sylvestre) [reply src/uu/fold/src/fold.rs:102]: The CI reported it

**Comment by sylvestre** [comment]
Many thanks!

**Pull request merged**